landscape/client/broker/tests/test_exchange.py

```diff
@@ -324,6 +324,89 @@ def exchange_callback():
         self.assertEqual(payload["sequence"], 1)
         self.assertEqual(payload["next-expected-sequence"], 0)
 
+    @mock.patch("landscape.client.broker.store.MessageStore"
+                ".delete_old_messages")
```

Comment:
Here I am trying to specifically test that store.delete_old_messages() gets called (messages.py line 39), because if it doesn't then old messages will be sent again which fails test_payloads_when_next_expected_too_high. This way it's clear that it happens because that call was missing.